### ISSUE TITLE
mcptam build fails with catkin build, works with catkin_make

### ISSUE THREAD

**Issue by stevenwaslander** [wavelab/mcptam#28]
Error message is:

```
CMake Error at CMakeLists.txt:15 (find_package):
By not providing "FindSUITESPARSE.cmake" in CMAKE_MODULE_PATH this project
has asked CMake to find a package configuration file provided by
"SUITESPARSE", but CMake did not find one.

Could not find a package configuration file provided by "SUITESPARSE" with
any of the following names:

  SUITESPARSEConfig.cmake
  suitesparse-config.cmake

Add the installation prefix of "SUITESPARSE" to CMAKE_PREFIX_PATH or set
"SUITESPARSE_DIR" to a directory containing one of the above files.  If
"SUITESPARSE" provides a separate development package or SDK, be sure it
has been installed.
```

**Comment by StanJBrown** [comment]
That's due to the cmakelists file. I had the same problem when installing.
Ill take a look tonight although I'm not sure how much uploading I can do.

**Comment by stevenwaslander** [comment]
If you email me a fix, I can fiddle with it.  I already have a
    branch for fixing it.  Trying to learn something new everytime I
    find something broken...
    Steve
    
    On 15-12-18 04:34 PM, Stanley Brown
      wrote:
    
    
      
      That's due to the cmakelists file. I had the same problem when
      installing.
      Ill take a look tonight although I'm not sure how much uploading I
      can do.
      On Dec 18, 2015 1:22 PM, "Steven Waslander"
      <notifications@github.com>
      wrote:
      
      > Assigned #28
      <https://github.com/wavelab/mcptam/issues/28> to @StanJBrown
      > <https://github.com/StanJBrown>.
      >
      > —
      > Reply to this email directly or view it on GitHub
      >
      <https://github.com/wavelab/mcptam/issues/28#event-496265197>.
      >
      —
        Reply to this email directly or view
          it on GitHub.

**Comment by StanJBrown** [comment]
The error is due to the cmake path to the CMAKEmodules folder located in
mcptam/cmake_modules folder being set correctly when using catkin build. To
get around it I changed a line in the root cmake file. This file is located
in catkin_ws/mcptam/CMakeLists.txt.

Go into it and go to line 4 or 5 and change it to the following the line
that says "set(CMAKE_MODULE_PATH .... "

to
set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH}
"~/catkin_ws/src/mcptam/cmake_modules")

or whatever your catkin_ws folder name is. Basically you need to hardcode
the location of the cmake_modules folder when using catkin build for some
reason. So that's the hack to get around it. Now we need figure out why it
happens only with catkin build and how to fix it as this will not work for
any type of deployment.

I will try see what I can find on the ros forums.

Cheers,
Stan

**Comment by stevenwaslander** [comment]
I can confirm this is the case.  Hard coded CMAKE_MODULE_PATH to "~/catkin_ws/src/mcptam/cmake_modules", same dir in my install, and it all worked.  Without it, failure. 

Something wrong with setting the path this way. Either CMAKE_SOURCE_DIR or PROJECT_NAME not getting set correctly.

```
set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/${PROJECT_NAME}/cmake_modules")
```

**Comment by stevenwaslander** [comment]
Closed with pull request #29
